server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -163,16 +165,15 @@ public boolean upgradeNode(String downloadJSON) {
     @Override
     public TestSession getNewSession(Map<String, Object> requestedCapability) {
         LOGGER.entering();
-        TestSession session;
+        //verification should be before synchronized block also to avoid unnecessarily acquiring lock
+        if (totalSessionsStarted >= maxSessionsAllowed || forceShutDown) {
+            return logSessionInfo();
+        }
         synchronized (this) {
             if (totalSessionsStarted >= maxSessionsAllowed || forceShutDown) {
```

Comment:
That makes sense.  we may want to add a comment somewhere on the 2nd if test.  Also don't we need to now make forceShutDown volatile also if the first test is outside the lock?